WordPress/Classes/ViewRelated/Views/WPRichText/WPRichTextFormatter.swift

```diff
@@ -119,12 +119,24 @@ class WPRichTextFormatter {
             let attachmentString = NSMutableAttributedString(attributedString: NSAttributedString(attachment: attachment))
             attachmentString.addAttributes(attributes, range: NSRange(location: 0, length: attachmentString.length))
 
+            if needsAppendNewLine(to: str, at: range) {
+                attachmentString.append(NSAttributedString(string: "\n"))
+            }
+
             attrString.replaceCharacters(in: range, with: attachmentString)
         }
 
         return attrString
     }
 
+    func needsAppendNewLine(to string: String, at nsRange: NSRange) -> Bool {
```

Comment:
The method name is kind of generic but its solving a specific glitch.  Since this is specific to horizontal rules, would it make sense to tweak the method name to reflect that?